zebra-consensus/src/primitives/groth16/hash_reader.rs

```diff
@@ -0,0 +1,43 @@
+use std::io::{self, Read};
+
+use blake2b_simd::State;
+
+/// Abstraction over a reader which hashes the data being read.
```

Comment:
Don't we already have an impl of `HashReader` somewhere?

If not, this seems like generic utility code, rather than groth16-specific code, so it belongs somewhere else in the module tree.

Reply:
I think an equivalent is https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/serialization/sha256d.rs, but that's for SHA256d, this one is for Blake2b; we can more it to a top level module still in `zebra-consensus`, or `zebra-chain`.